tests/50federation/40devicelists.pl

```diff
@@ -202,20 +202,10 @@
 
 
 test "Server correctly resyncs when client query keys and there is no remote cache",
-   requires => [ local_user_fixture(),
-                 $main::INBOUND_SERVER, $main::OUTBOUND_CLIENT,
-                 federation_user_id_fixture(),
-                 room_alias_name_fixture() ],
+   requires => [ $main::INBOUND_SERVER, federated_rooms_fixture() ],
 
    check => sub {
-      my ( $user, $inbound_server, $outbound_client, $creator_id, $room_alias_name ) = @_;
-
-      my ( $room_id );
-
-      my $remote_server_name = $inbound_server->server_name;
-      my $datastore          = $inbound_server->datastore;
-
-      my $room_alias = "#$room_alias_name:$remote_server_name";
+      my ( $inbound_server, $user, $federated_user_id ) = @_;
```

Comment:
this probably works because you don't care about the room_id, but it's hiding the fact that there is a room_id that you are throwing away.

```suggestion
      my ( $inbound_server, $user, $federated_user_id, undef) = @_;
```